src/main/java/edu/njit/jerse/ashe/services/MethodReplacementService.java

```diff
@@ -202,53 +198,29 @@ private static List<String> splitParameters(String parameterString) {
-     * format of a valid Java method signature.
-     *
-     * @param signature the method signature to be checked
-     * @return {@code true} if the method signature is complete and a valid Java method signature; {@code false} otherwise.
-     */
-    private static boolean isValidMethodSignature(MethodSignature signature) {
-        boolean isValidSig =
-                signature.returnType() != null &&
-                        signature.methodName() != null &&
-                        signature.parameters() != null;
-
-        if (isValidSig) {
-            LOGGER.debug("Java method signature is valid: returnType={}, methodName={}, parameters={}",
-                    signature.returnType(), signature.methodName(), signature.parameters());
-        } else {
-            LOGGER.warn("Invalid Java method signature detected: returnType={}, methodName={}, parameters={}",
-                    signature.returnType(), signature.methodName(), signature.parameters());
-        }
-
-        return isValidSig;
-    }
-
     /**
      * Retrieves the declaration of a specific class or interface by name from the
      * provided compilation unit.
      *
-     * @param cu        The compilation unit from which the class or interface
-     *                  declaration needs to be extracted.
-     * @param className The name of the class or interface whose declaration is
-     *                  to be fetched.
-     * @return An {@code Optional} containing the declaration of the target
-     * class or interface if found; otherwise, an empty {@code Optional}.
+     * @param cu        the compilation unit from which the class or interface
+     *                  declaration needs to be extracted
+     * @param className the name of the class or interface whose declaration is
+     *                  to be fetched
+     * @return the declaration of the target class or interface.
      */
-    private static Optional<ClassOrInterfaceDeclaration> getClassDeclaration(CompilationUnit cu, String className) {
+    private static ClassOrInterfaceDeclaration getClassDeclaration(CompilationUnit cu, String className) {
         List<ClassOrInterfaceDeclaration> classes = cu.findAll(ClassOrInterfaceDeclaration.class);
 
         for (ClassOrInterfaceDeclaration classOrInterface : classes) {
             if (classOrInterface.getNameAsString().equals(className)) {
                 LOGGER.debug("Retrieved the targeted class declaration: {}", className);
-                return Optional.of(classOrInterface);
+                return classOrInterface;
             }
         }
 
-        LOGGER.warn("The targeted class declaration was not found in the provided compilation unit.");
-        return Optional.empty();
+        errorMessage = "The targeted class declaration was not found in the provided compilation unit.";
+        LOGGER.warn(errorMessage);
+        throw new IllegalArgumentException(errorMessage);
```

Comment:
missing `@throws` clause and `throws IllegalArgumentException` on the method signature.